### ISSUE TITLE
Error spawning child process

### ISSUE THREAD

**Issue by cvanelteren** [balena-io/etcher#4274]
When trying to flash the archiso using 2:1.19.21-3 I was faced with the error.

![image](https://github.com/balena-io/etcher/assets/19485143/6eb4865a-baa8-43df-8468-56e866cfa18d)

- **Etcher version:** 
- **Operating system and architecture:** 
- **Image flashed:** 
- **What do you think should have happened:** 
- **What happened:**
- **Do you see any meaningful error information in the DevTools?**

**Comment by sukanka** [comment]
Hey, it seems you are using `balena-etcher` from AUR. If so, please rebuild it and try again.

**Comment by cvanelteren** [comment]
Hi @sukanka! I tried several versions offered through AUR without success. 
- etcher-bin
- balena-etcher
- etcher-ng-bin

With the etcher-git version a different error pops up 
![image](https://github.com/balena-io/etcher/assets/19485143/2dce27f5-5dfa-452b-9b85-b384530ebd64)
With rebuilding it, are you referring to do this directly from release page?

**Comment by sukanka** [comment]
> Hi @sukanka! I tried several versions offered through AUR without success.
> 
> * etcher-bin
> * balena-etcher
> * etcher-ng-bin
> 
> With rebuilding it, are you referring to do this directly from release page?

No, I  mean rebuild `balena-etcher` from the source, just clone its AUR repo and run `makepkg` or `extra-x86_64-build` in the directory. 

Please also make sure  you use the same version of `nodejs` as you build it in the chroot

**Comment by Perlovka** [comment]
Same is happening with prebuilt app (linux-x64) from the site. As far as I understand, etcher is completely unusable for now. Gazillion bugs opened and zero fixes for more than a month. RIP.

**Comment by binba2** [comment]
Getting the same exact thing, from the prebuilt (macos-x64). MacOS 12.7.2.

**Comment by cvanelteren** [comment]
I used rufus in the mean time to burn the iso. Works like a charm.

**Comment by Vyacheslav1557** [comment]
same for me

**Comment by hwac121** [comment]
Please fix this guys, my favorite and usually dependable burning app is no longer of any use at all.

**Comment by cvanelteren** [comment]
Also @sukanka building from source gave me the same error.

**Comment by sukanka** [comment]
> Also @sukanka building from source gave me the same error.

quite strange, I have tested it locally and it works just fine.

**Comment by z0rg0n** [comment]
I get the 'Error spawning child process' error as well.

I've downloaded and ran the following and they all gave me the same problem. 

- [balena-etcher_1.19.21_amd64.deb](https://github.com/balena-io/etcher/releases/download/v1.19.21/balena-etcher_1.19.21_amd64.deb)
- [balena-etcher_1.19.19_amd64.deb](https://github.com/balena-io/etcher/releases/download/v1.19.19/balena-etcher_1.19.19_amd64.deb)
- [balenaEtcher-linux-x64-1.19.21.zip](https://github.com/balena-io/etcher/releases/download/v1.19.21/balenaEtcher-linux-x64-1.19.21.zip)


I've tried it with multiple ISO files.

Trying to download [balenaEtcher-1.19.21-x64.AppImage](https://github.com/balena-io/etcher/releases/download/v1.19.21/balenaEtcher-1.19.21-x64.AppImage) and just running that gives a different error (Error: (0 , h.requestMetadata) is not a function) when trying to add an ISO link and .


System info:
6.1.0-23-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.99-1 (2024-07-15)

**Comment by Keeqler** [comment]
On ArchLinux, I got the same error because I didn't have an authentication agent (like `polkit-kde-agent`) installed and running.
Etcher needs this so it can prompt for your password.

**Comment by z0rg0n** [comment]
It looks like polkit-kde-agent is already part of Debian?
https://packages.debian.org/bullseye/polkit-kde-agent-1

Anyway if anyone is just looking for a work around I was able to get it the thumbdrive flashed using dd in Debian as showing here:
https://linuxize.com/post/create-bootable-debian-10-usb-stick-on-linux/

**Comment by gitAcceptance** [comment]
@cvanelteren did you try running it as administrator?

**Comment by cvanelteren** [comment]
> @cvanelteren did you try running it as administrator?

No as it is not supported by electron. 
Installing the kde-pol-agent did not resolve it.

**Comment by wpf99** [comment]
I solved it by making my account an administrator on Mac. I had been a normal user and would use the admin account for installs, etc. when prompted for the use password I put in my password and would receive this error message. I went to Users and Groups and promoted myself to administrator and the problem is solved. It seems like this error is essentially: "not in sudoers group..."

**Comment by cvanelteren** [comment]
> [0913/203740.556208:FATAL:electron_main_delegate.cc(295)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.

**Comment by juniorr001** [comment]
> When trying to flash the archiso using 2:1.19.21-3 I was faced with the error.
> 
> ![image](https://private-user-images.githubusercontent.com/19485143/347318417-6eb4865a-baa8-43df-8468-56e866cfa18d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NzQzOTgsIm5iZiI6MTcyODY3NDA5OCwicGF0aCI6Ii8xOTQ4NTE0My8zNDczMTg0MTctNmViNDg2NWEtYmFhOC00M2RmLTg0NjgtNTZlODY2Y2ZhMThkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDE5MTQ1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3NDc3NzA0NTk1NWE5MThjMDIxMDY3NTJhNWZjNmUzMGJmNzExYTMxZmVmNDIxM2FkMzk1ZGNlMjk5ZTg5NDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.chUe6ibKbV2LT-WJ-5L-8SUhIuxiou_s0TToK6utcQ8)
> 
> * **Etcher version:**
> * **Operating system and architecture:**
> * **Image flashed:**
> * **What do you think should have happened:**
> * **What happened:**
> * **Do you see any meaningful error information in the DevTools?**

I solved it by Running as administrator.

**Comment by cvanelteren** [comment]
Yes, but it cannot find the monitor then on wayland

**Comment by bps86** [comment]
On X11, Usually the way to fix this is to run it with sudo without sandbox. The problem is that this method is impossible on wayland because it's kind of hard to make sudo work with GUI apps.

**Comment by gitAcceptance** [comment]
Wow, a rare win for windows.

**Comment by thegoodduck** [comment]
Hey me too, now using GNOME Disks

**Comment by IPlayZed** [comment]
This happens as well on me on Windows 11 23H2, I do not see any Windows Defender related things.

**Comment by achodeveloper2** [comment]
Def sounds like a security flaw.

**Comment by fatso83** [comment]
Related, just with more debug logs, screenshots, ++: #4356

**Comment by Temperature-block** [comment]
I was having same issue i ran balena via ssh with x11 forwarding i was able to resolve this issue by runnng balena from xfce4-panel

**Comment by JadenB14** [comment]
I'm using Hyprland and was getting this same error, but I downloaded hyprpolkitagent-git  from AUR and when running it with `/usr/lib/hyprpolkitagent` in a terminal, it started the agent and allowed for balena-etcher to ask for my user password. Without a polkit agent installed and **running**, it cannot spawn the child process necessary because of limited permissions.

You should be able to get a compatible polkit agent for your OS, just remember that it doesnt start automatically, look and see how to manually active it.

**Comment by JadenB14** [comment]
@cvanelteren So you downloaded the polkit-kde-agent from the AUR? If so, to activate your poling agent type, `/usr/lib/polkit-kde-authentication-agent-1` into an empty terminal, then open balena-etcher and try your flash again.

**Comment by cvanelteren** [comment]
Will check!

**Comment by cvanelteren** [comment]
Can confim issue is gone after installing `hyprpolkitagent`. Closing!

**Comment by aentwist** [comment]
Thanks for the helpful information, but can someone explain this? I think I have to go elsewhere for now.

Etcher 1.19.25 from deb package, Debian 12

```
$ apt search polkit
polkit-kde-agent-1/stable,now 4:5.27.5-2 amd64 [installed,automatic]
  KDE dialogs for PolicyKit

polkitd/stable,now 122-3 amd64 [installed,automatic]
  framework for managing administrative policies and privileges
```

```
$ systemctl status polkit
● polkit.service - Authorization Manager
     Loaded: loaded (/lib/systemd/system/polkit.service; static)
     Active: active (running)
```

**Comment by IPlayZed** [comment]
@aentwist I am pretty sure you can get verbose information out of the polkit agent, maybe in the system journal. Can you share that when trying to use Balena?

**Comment by aentwist** [comment]
Yeah, true. It doesn't appear to be called at all. That is, looks empty to me. I do have some Electron experience, so maybe I'll be able to check and see one day, but I doubt it at this time. I used dd successfully for my need this time. Background: fresh Debian + KDE Plasma (not GNOME) installation.

**Comment by yPhil-gh** [comment]
Still broken here (.deb, latest Ubuntu).
Oh, and the AppImage is broken, too.

```
➜ ~/bin/balenaEtcher-1.7.9-ia32.AppImage             
dlopen(): error loading libfuse.so.2

AppImages require FUSE to run. 
You might still be able to extract the contents of this AppImage 
if you run it with the --appimage-extract option. 
See https://github.com/AppImage/AppImageKit/wiki/FUSE 
for more information

```
This is not even true. I run AppImages all the time. Oh well, it was good while it lasted, b-by Etcher 😢

**Comment by timbostop** [comment]
I have same. On a clean install on Mac OS x64. Only way to run the app successfully is as an admin user (you can't just enter an admin password). 

But it doesn't tell you this - only "failed to spawn child process" once you flash.

It's definitely not "flawless"

**Comment by Kabilan108** [comment]
Having the same issue on Ubuntu 24.04 LTS. Have tried installing via the amd64 deb package but I still get the same error

**Comment by jashwanth450** [comment]
all you need to do is Run as Admin. Works for me.

**Comment by emece67** [comment]
Just downloaded the .deb file from github, installed it on Debian+i3 with `apt` and got the same error. Running it with sudo doesn't help, also the same error.

**Comment by RochaA4** [comment]
I installed it using the zip(64) file off the balena website, unzipped the file, opened 2 Terminals, ran /usr/lib/polkit-kde-authentication-agent-1 and ./balena in those separate terminals. Im using Arch and Hyprland.

**Comment by Victor-N-Suadicani** [comment]
Getting the same error. I get a prompt to input my password, but then it freezes and later shows this error. This should be reopened.
